runner.py

```diff
@@ -0,0 +1,495 @@
+#!/usr/bin/env python
+
+import sys
+assert sys.version_info >= (3, 9), "Use Python 3.9 or newer."
+
+import os
+import socket
+import optparse
+import datetime
+import shutil
+import subprocess
+import time
+import signal
+import json
+from typing import Union, Tuple
+
+'''
+The default flags depending on the hostname.
+'''
+DEFAULT_SETTINGS = {
+    'futharkhpa01fl.unicph.domain': {
+        'name': 'A100',
+        'labels': 'cuda,opencl,multicore',
+    },
+    'futharkhpa02fl.unicph.domain': {
+        'name': 'MI100',
+        'labels': 'opencl,multicore',
+    },
+}
+
+'''
+These are the commands used for installing the self-hosted runner. You may wish to update these
+commands with commands from the download box on the website.
+https://github.com/{organization}/{repository}/settings/actions/runners/new
+'''
```

Comment:
Why is this a string?  Are Python comments not good enough anymore?

Reply:
Oh, is it a docstring?  I don't know how Python does this.

Reply:
I did a tiny bit of research and now I am unsure if constants work with __doc__ or any other ways of doing documentation in python.